modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -205,6 +205,71 @@
 					title: editor.lang.link.title,
 				});
 
+				/**
+				 * Spacing options are hardcoded to 1.0x, 1.5x, 2.0x, 3.0x, 4.0x and 5.0x,
+				 * so that an empty class, “mt-1 mb-1” class, up to “mt-5 mb-5” are applied respectively.
```

Comment:
Why don't we  use a `CKEDITOR.style` with the `line-height` property set to whatever option was selected? (That would avoid having to define CSS classes)

Reply:
no problem, I think this is "kind of" related to this [comment](https://github.com/liferay-frontend/liferay-portal/pull/1150#discussion_r647330117), so I'll  ping @edalgrin as well here... since CKEditor transformations are mostly done inline, is there any reason we want to associate this with CSS classes?

Reply:
Not sure if we are defining new classes or using Clay but to use the CSS Custom Properties, _propertly 😄_ we should have something like:

```css
[class*="mt-"] {
   margin-top: var(--spacer); // !important; ?
   line-height: var(--spacer);
}
[class*="mb-"] {
   margin-bottom: var(--spacer);
   line-height: var(--spacer);
}
...
.mt-1 {
   --spacer: 1rem;
}
.mt-2 {
   --spacer: 2rem;
}
.mt-3 {
   --spacer: 3rem;
}
...
```

Actually we should apply this to Clay too, it should save us many KB of CSS 🤔 (not sure about the new `ms` and `me` naming stuff https://getbootstrap.com/docs/5.0/utilities/spacing/#notation)

Reply:
@edalgrin I don't understand why custom properties are needed to change the line height of an element

Reply:
@julien I don't know 🤔  I was looking at 
> "line-height: var(--spacer-".concat(value, ") !important"

and how to change it "without" JS, but then I noticed
> That would avoid having to define CSS classes

so.. nevermind? 😆 

sorry I think I misunderstood the request

Reply:
Ideally, BE should just set classes on the affected elements as a result of selecting any option in the spacing menu. Same considerations would apply in other dropdown menus related to formatting items. The improvement over AE is that the classes to use and the dropdown menu entries are configurable. In case no config is given, we'll keep old AE defaults.

This way:
* BE does not need to know about custom properties, keeping it simple and backporteable
* BE users don't need to add extra configurations to get what they had with AE, thanks to the defaults
* Still, class definitions may be based on custom properties so whatever classes are set by the editor, the edited content could still benefit from tokens and stylebooks if the theme in question decides to use them

Does that make sense to you @diegonvs ?

Reply:
Hey @dsanz, check https://github.com/liferay-frontend/liferay-portal/pull/1150#issuecomment-859963910. 

Make it configurable section